File: core/src/engine/general/view/gui/IconLabel.scala
package engine.general.view.gui

import java.awt.{Image, Graphics}
import java.util.Locale
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.graphics.glutils.ShapeRenderer

object IconLabel{
	class Builder(){
		var x:Int=0
		var y:Int=0
		var image:Texture=null
		var h:Int=0
		var w:Int=0
		
		def xPos(xP:Int):Builder={
			x=xP
			return this
		}
		
		def yPos(yP:Int):Builder={
			y=yP
			return this
		}
		
		def image(img:Texture):Builder={
			image=img
			return this
		}
		
		def height(ht:Integer):Builder={
			h=ht
			return this
		}
		
		def width(wh:Integer):Builder={
			w=wh
			return this
		}
		
		def size(sz:Integer):Builder={
			h=sz
			w=sz
			return this
		}
		
		def build():IconLabel=return new IconLabel(this)
	} 
}

/**
 * This object is a container for storing images.
 */
class IconLabel(xPos:Integer,yPos:Integer,val builder:IconLabel.Builder) extends GuiItem(xPos,yPos){
    
	val height=builder.h
	val width=builder.w
	val image=builder.image 
	
	def this(builder:IconLabel.Builder){
		this(builder.x,builder.y,builder)
	}
	
    val drawX=xPos.toFloat
    val drawY=yPos.toFloat

    def draw(render:ShapeRenderer,batch:SpriteBatch){
    	batch.begin()
    	batch.draw(image, drawX, drawY, width,height)
    	batch.end()
    }
}

File: core/src/engine/general/view/gui/IconLabel.java
package engine.general.view.gui;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.glutils.ShapeRenderer;

public class IconLabel extends GuiItem{
  
	public static class Builder{
		
		int x=0;
		int y=0;
		Texture image=null;
		int h=0;
		int w=0;
		
		public Builder xPos(int xP){
			x=xP;
			return this;
		}
		
		public Builder yPos(int yP){
			y=yP;
			return this;
		}
		
		public Builder image(Texture img){
			image=img;
			return this;
		}
		
		public Builder height(Integer ht){
			h=ht;
			return this;
		}
		
		public Builder width(Integer wh){
			w=wh;
			return this;
		}
		
		public Builder size(int sz){
			h=sz;
			w=sz;
			return this;
		}
		
		public IconLabel build(){
			return new IconLabel(this);
		}
	} 
	
	final int height;
	final int width;
	final Texture image;
	    
	public IconLabel(Builder b){
		super(b.x,b.y);
	    height=b.h;
		width=b.w;
		image=b.image;
	}

    public void draw(ShapeRenderer render,SpriteBatch batch){
    	batch.begin();
    	batch.draw(image, xPos, yPos, width,height);
    	batch.end();
    }
}
